lib/resque/scheduler/lock/resilient_modern.rb

```diff
@@ -0,0 +1,14 @@
+# vim:fileencoding=utf-8
+require_relative 'base'
+
+module Resque
+  module Scheduler
+    module Lock
+      class ResilientModern < Resilient
```

Comment:
I couldn't come up with a better name

Reply:
What does inheriting from `Resilient` give us?

I'd prefer inheriting from `base` and add a implementation for `locked?`. This would make it easier to remove the `Resilient` (and `Basic`) lock implementations in the future, which I can't imagine we wouldn't do now that we have this awesome new lock implementation (and redis `2.x` is very old at this point).

Reply:
@iloveitaly The `Resilient` implementation offers the `locked?` method which retrieves and renews the lock.  So, it looks like in Redis 6.2 a [similar command](https://redis.io/commands/getex) was added.

@yaauie suggested I retain the existing implementation to avoid users downgrading to a worse master lock implementation if they upgraded resque-scheduler.

One possibility could be switching the Redis requirement to 6.2 and completely removing the inheritance and Lua script implementation.